### PR TITLE
Show warning on piece doctor page if still doing initial scan

### PR DESCRIPTION
**Pull request by dirkmc**
Fixes https://github.com/filecoin-project/boost/issues/1640

Screenshot 2023-08-23 at 2 54 15 PM

TODO:
- [x] Test in production

### CODE REVIEW

File: extern/boostd-data/client/client.go [hunk 1]
```diff
@@ -27,6 +27,7 @@ type Store struct {
 		GetOffsetSize             func(context.Context, cid.Cid, mh.Multihash) (*model.OffsetSize, error)
 		ListPieces                func(ctx context.Context) ([]cid.Cid, error)
 		PiecesCount               func(ctx context.Context, maddr address.Address) (int, error)
+		ScanProgress              func(ctx context.Context, maddr address.Address) (*types.ScanProgress, error)
```
Comment (nonsense) [inline extern/boostd-data/client/client.go:30]: The API for boostd-data has exploded quite a bit with auxiliary functionality.

I wonder if we can refactor in the future and split into 2 APIs -- one for critical functionality that one needs to implement if they want their own adapter for a new database (such as PostgreSQL with Citus), and one for auxiliary functionality that we build - Piece Doctor, UI feedback (such as ScanProgress), etc.

Just a thought as we keep on adding new methods here, no action required at this time.

File: extern/boostd-data/yugabyte/piecedoctor.go [hunk 2]
```diff
@@ -235,6 +235,10 @@ func (s *Store) getPieceCheckPeriod(ctx context.Context) (time.Duration, error)
 	return period, nil
 }
 
+// Returns the number of rows in the PieceTracker table.
```
Comment (nonsense) [inline extern/boostd-data/yugabyte/piecedoctor.go:238]: Nit but for go-doc it is convention to start the comment with the name of the method (i.e. `PieceCount returns the number...`

If we are adding go-doc to some/all methods, we might as well stick to the convention.

File: extern/boostd-data/yugabyte/piecedoctor.go [hunk 3]
```diff
@@ -249,6 +253,53 @@ func (s *Store) PiecesCount(ctx context.Context, maddr address.Address) (int, er
...
+	// Calculate approximate progress
+	progress := float64(scanned) / float64(total)
+
+	// Given that the denominator may be a little inflated, round up to 100% if we're close
+	if progress > 0.95 {
+		progress = 1
+	}
+	if progress > 1 {
+		progress = 1
+	}
```
Comment (nonsense) [inline extern/boostd-data/yugabyte/piecedoctor.go:298]:
```suggestion
```

**Review by nonsense — APPROVED** [review]

File: extern/boostd-data/client/client.go (hunk 1 above)
Comment (dirkmc) [reply extern/boostd-data/client/client.go:30]: That's true, we are adding a bunch of new methods 👍

File: extern/boostd-data/yugabyte/piecedoctor.go (hunk 2 above)
Comment (dirkmc) [inline extern/boostd-data/yugabyte/piecedoctor.go:238]:
```suggestion
// PiecesCount returns the number of rows in the PieceTracker table.
```

**Pull request merged**